app/scripts/modules/amazon/src/loadBalancer/configure/common/targetGroupValidators.spec.ts

```diff
@@ -0,0 +1,162 @@
+import * as Validators from './targetGroupValidators';
+
+const mockTargetGroup = {
+  attributes: {
+    deregistrationDelay: 600,
+    stickinessDuration: 8400,
+    stickinessEnabled: false,
+    stickinessType: 'lb_cookie',
+  },
+  healthCheckInterval: 10,
+  healthCheckPath: '/healthcheck',
+  healthCheckPort: 7001,
+  healthCheckProtocol: 'HTTP',
+  healthCheckTimeout: 5,
+  healthyThreshold: 10,
+  name: 'targetgroup',
+  port: 7001,
+  protocol: 'HTTP',
+  targetType: 'ip',
+  unhealthyThreshold: 2,
+  account: 'test',
+  cloudProvider: 'aws',
+  healthTimeout: 1000,
+  healthInterval: 10,
+  loadBalancerNames: ['loadbalancer1', 'loadbalancer2'],
+  region: 'us-east-1',
+  type: '',
+};
+
+describe('Target Group validators', () => {
+  describe('of used names', () => {
+    it('returns an error if the name exists already', () => {
+      const existingGroups = {
+        test: {
+          'us-east-1': ['targetgroup'],
+        },
+      };
+      const actual = Validators.isNameInUse(existingGroups, 'test', 'us-east-1')(mockTargetGroup.name);
+      const expected = 'There is already a target group in test:us-east-1 with that name.';
+      expect(actual).toEqual(expected);
+    });
+
+    it('returns an null if it does not exist', () => {
+      const existingGroups = {
+        test: {
+          'us-east-1': ['targetgroup2'],
+        },
+      };
+      const actual = Validators.isNameInUse(existingGroups, 'test', 'us-east-1')(mockTargetGroup.name);
+      expect(actual).toEqual(null);
+    });
+  });
+
+  describe('of name length', () => {
+    it('returns error if the name is >32 additional characters', () => {
+      const actual = Validators.isNameLong(12)(`${mockTargetGroup.name}areallylongname`);
```

Comment:
minor testing nit: In assertions like this, it's easier to understand the intention of the test if you use literal values instead of references, even if it means more "duplication".  For example, I think this assertion would be easier to grok if it was:

```suggestion
      const actual = Validators.isNameLong('application13'.length)('areallylongtargetgroupname');
```

This way, you don't have to wonder what `12` means, nor what is in `mockTargetGroup.name`